lib/image-stream/parse-partitions.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const Bluebird = require('bluebird');
+const MBR = require('mbr');
+const GPT = require('gpt');
+
+const MAX_BYTES = 8192;
+const INITIAL_LENGTH = 0;
+const INITIAL_BLOCK_SIZE = 512;
+const MAX_BLOCK_SIZE = 4096;
+
+/**
+ * Attempt to parse the GPT from various block sizes
+ * @private
+ * @param {Buffer} buffer - Buffer
+ * @returns {GPT|null}
+ */
+const parseGPT = (buffer) => {
+
+  let blockSize = INITIAL_BLOCK_SIZE;
```

Comment:
Definitely blockSize, yes – because the GPT should always be at LBA 1 (the MBR at LBA 2) – and doesn't have a GPT specific offset, it's always connected to the block size